### DISCUSSION TITLE
iiRcade port

### DISCUSSION THREAD

**Discussion by SparrowRex** [diasurgical/DevilutionX#6477] [Ideas]
The iiRcade has recently been unlocked and can have software pushed to it for those that have the USB enabler. It has an Android OS and so far can have MAME and Android games pushed to it.

Was wondering if we can have the awesome devilutionX.

**Comment by AJenbo** [comment]
Have you tried installing the Android version?

**Reply by SparrowRex** [reply]
Not yet. I have the USB enabler on order. Once I get it then I can test it out and report back. I know there is control mapping involved. Not too familiar just yet.

**Reply by SparrowRex** [reply]
- Update: Received the USB Enabler.
- MIA software allows for me to build the software package to push the game to the iiRcade system.
- After trial and error I was able to figure a config that aligns (Arcade Buttons) with the gamepad mapping within devilutionx.
- Works fantastic!! My next part of the process is to add the background image and instructions image for the iiRcade console.
- There is an iiRcade community discord for sharing packs and they encourage devs to support the system. Not sure if an official package is wanted here. I can always share mine. Devs have been using itch.io to share software also.

**Reply by StephenCWills** [reply]
> Not sure if an official package is wanted here. I can always share mine.

What is an official package in this case? MIA is used to sideload APKs right? Does it produce another format that includes the button configurations or something?

**Reply by SparrowRex** [reply]
Correct with MIA it allows the sideloading but needs an "iiGames" package created. In this package is the apk, Button Configuration, Background image, and instruction image. So I guess official package would be config, images, and latest version of the apk.

Also here is the link explaining MIA:
[https://www.team-encoder.com/MIA](https://www.team-encoder.com/MIA)

Here is an itch.io collection that someone put together showing iiRcade dev support:
[https://itch.io/c/3657380/iircade-games-for-sideloading](https://itch.io/c/3657380/iircade-games-for-sideloading)

**Reply by AJenbo** [reply]
Any pictures or video of the game running on the system?

**Comment by SparrowRex** [comment]
Here are some pictures. Looks like video I'll need to have less than 10MB.
You can see the controls are labeled A,B,C  X,Y,Z. I matched it up to the game Pad Mapping default with C = LB, and Z = RB; P1 Start = Start; Coin = Select.
Plays Awesome!! Without needing to use in-game control mapping.

![iiRcade_Menu](https://github.com/diasurgical/devilutionX/assets/142461808/35b963dd-d776-4438-b619-a59825422ff4)
![iiRcade_DB_InGame](https://github.com/diasurgical/devilutionX/assets/142461808/a6c5baa0-75f5-48ee-84b1-4c08c58013d3)
![iiRcade_DB_Menu](https://github.com/diasurgical/devilutionX/assets/142461808/713cb523-4374-4e81-b86d-05198efceafa)

**Reply by AJenbo** [reply]
Nice, thanks for sharing :)

P.s. 1.5.1 is available ;)

**Reply by SparrowRex** [reply]
Thank you for the fine work! :-)
